GliaWidgets/ViewModel/Chat/ChatViewModel.swift

```diff
@@ -368,26 +369,37 @@ extension ChatViewModel {
                     dismissed: nil
                 )
             }
-        } else {
-            let messageItem = ChatItem(with: outgoingMessage)
+
+        case .enqueued:
+            handle(pendingMessage: outgoingMessage)
+
+        case .enqueueing, .ended, .none:
+            handle(pendingMessage: outgoingMessage)
+            enqueue(mediaType: .text)
```

Comment:
Why do we need to call enqueue again here? what if it is a call?

Reply:
According to our previous and current logic we need to establish engagement if user sends message if it isn't already established. There I only replaced `if  interactor.isEngaged` with `switch` and added  separate handling for `enqueued` state.

Reply:
The previous logic was: until interactor tells us we are not engaged, add messages to pending queue. Once interactor tells us that we are engaged, send all the messages until queue is empty.
Calling enqueue(mediaType: .text) without checking whether the chat was actually supposed to start engagement or not (since Audio/Video call also starts chat) can lead to some odd behaviour I think.

Reply:
Looks like you cant go to chat anyway when call is in queue, although I don't see why we would not allow it. Might be some case we haven't covered yet but should.